modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/util/DLBreadcrumbUtil.java

```diff
@@ -143,7 +146,16 @@ private static String _getRootFolderName(
 		Group group = themeDisplay.getScopeGroup();
 
 		if (repositoryId != 0) {
-			group = GroupServiceUtil.getGroup(repositoryId);
+			try {
+				group = GroupServiceUtil.getGroup(repositoryId);
+			}
+			catch (PortalException portalException) {
+				_log.error(portalException);
+
+				if (folder != null) {
+					group = GroupServiceUtil.getGroup(folder.getGroupId());
+				}
+			}
```

Comment:
When the repositoryId is not a groupId, you should take the group from the repository instead of using the folder (as the folder could be null).
```
			try {
				group = GroupServiceUtil.getGroup(repositoryId);
			}
			catch (PortalException portalException) {
				_log.error(portalException);

				Repository repository = RepositoryServiceUtil.getRepository(repositoryId);
				group = GroupServiceUtil.getGroup(repository.getGroupId());
			}
```

Reply:
But in this case, the repository.getGroupId will be (usually) the liferay group, so it will not be the repository where the folderId is 🤔